### ISSUE TITLE
Getting yesterday's message Labelled as Today

### ISSUE THREAD

**Issue by Himesh27** [signalapp/Signal-Android#10745]
I have got notification today that I have received message from my friend but when I opened it, it showed that message was sent yesterday but it was labelled under "Today". Please look into it.
![Screenshot_2021-01-20-15-33-45-907_org thoughtcrime securesms](https://user-images.githubusercontent.com/12909112/105160265-0bd80480-5b36-11eb-84c1-5fb5a8ebf7d9.jpg)

**Comment by Leptopoda** [comment]
Signal will display messages based on when your phone receives it but display the sent time based on when the message was sent

Otherwise a malicious client could sent you a message and tell that the message is from one year ago. The message should not be automatically saved as one year ago as this would change your message history (I could send you a message today but tell the client it was a year ago. When you now scroll back to a year ago suddenly there is a message actually sent today. This would alter the message history which shouldn't be possible)
Also the time of phones never are 100% in sync. Just like your friends phone might have the time set to 5min ago. Sending you a message would display it in the past without you noticing.

thats the reason why your signal client displays the messages in order (and time) they where received but with the timestamp they actually where sent. 

Long story short this behavior is expected and not a bug. If you want to discuss this further (like you want that changed) i suggest you to have a look in our [community forum](https://community.signalusers.org) which is tailored to feature requests.
Would you please close this issue?

**Comment by Trombonista** [comment]
On S4mini V5.3.1 and previous versions something is different to Leptopodas decription: received messages have timestamp "now" despite notification came in some while ago.
